### PR TITLE
Add script to set up Python without conda

### PR DESCRIPTION
**Pull request by astrofrog**
Since the core package now uses choco to install Python on Windows and we might want to also avoid conda on MacOS X, I thought it would be handy to make a script that does all the set-up so we can use it across the organization as needed (the CI for the pytest plugins is having the same issues on Windows).

I am testing this out in https://github.com/astropy/astropy/pull/10871

### Why not use Homebrew on Mac?

It is quite slow and seems to download half the internet in some form of dependency hell.

### Why not use pyenv on Mac?

It downloads and builds Python from source so is very slow.

### Why use venv?

This is the easiest way to make sure we don't run into permissions issues and also makes available a ``python`` alias.

### Do we have to hard-code the bugfix mapping for MacOS X?

I could probably write a small script to try and do this automatically, but didn't want to spend too much time if we don't care.

### CODE REVIEW

**Comment by astrofrog** [comment]
@bsipocz - does this seem ok to you broadly speaking? If so will work on it some more and add docs etc to wrap up.

File: travis/setup_python.sh
Comment (bsipocz) [inline travis/setup_python.sh:29]: is the plan here to regularly revisit these numbers and update?

**Review by bsipocz — COMMENTED** [review]
overall looks good

File: travis/setup_python.sh
Comment (astrofrog) [reply travis/setup_python.sh:29]: I can write a small script to determine these automatically so we don't have to, but didn't want to spend time on it if you disagreed with it overall.

Comment (bsipocz) [reply travis/setup_python.sh:29]: (It's OK if it's, but then maybe it's time to keep a tally of such updatable stuff (the current ones I know about are: latest versions of hard wired things in `LATEST...`, this python version, conda version)

Comment (bsipocz) [reply travis/setup_python.sh:29]: I think it's reasonable enough to come back to it regularly, less sudden surprises that way (similar as with the conda verison)

**Comment by astrofrog** [comment]
@bsipocz I added some docs including a list of variables to update regularly.

**Comment by astrofrog** [comment]
https://github.com/astropy/astropy/pull/10871 seems to work, so ready for review

**Comment by pllim** [comment]
I thought that we're slowly moving away from ci-helpers, given Linux jobs doesn't need ci-helpers anymore. Seems counter intuitive to add more stuff here just for Windows and OSX.

**Comment by astrofrog** [comment]
To be clear we are moving away from conda ci helpers but we still want to use it for eg the pin file. Also I don't want to have to repeat this code in all Travis files :/

**Review by pllim — APPROVED** [review]
If this makes lives easier, why not.

Let's wait for @bsipocz 's approval before merge though.

**Comment by pllim** [comment]
With this in, maybe we can start seriously think about getting rid of Appveyor.

**Comment by astrofrog** [comment]
Yes I think since the appveyor script is broken anyway we could just display a message pointing people to using Windows on Travis instead.

File: README.md
Comment (bsipocz) [inline README.md:21]: maybe link to the section without conda in the first sentence after this?

**Review by bsipocz — APPROVED** [review]

File: README.md
Comment (astrofrog) [reply README.md:21]: Sounds good will add

Comment (pllim) [reply README.md:21]: @astrofrog , after adding it, I think we can merge? Also if you really like green status, rebase and you shall have it.

Comment (astrofrog) [reply README.md:21]: Done! Because master is a protected branch we do need to wait for the CI to run again though.

**Pull request merged**
